packages/nimble-components/src/rich-text/editor/tests/rich-text-editor-mention.spec.ts

```diff
@@ -0,0 +1,831 @@
+import { html } from '@microsoft/fast-element';
+import { richTextEditorTag, RichTextEditor } from '..';
+import { type Fixture, fixture } from '../../../utilities/tests/fixture';
+import { parameterizeNamedList } from '../../../utilities/tests/parameterized';
+import { RichTextEditorPageObject } from '../testing/rich-text-editor.pageobject';
+import { createEventListener } from '../../../utilities/tests/component';
+import {
+    RichTextMentionUsers,
+    richTextMentionUsersTag
+} from '../../../rich-text-mention/users';
+import { MappingUser, mappingUserTag } from '../../../mapping/user';
+import { waitForUpdatesAsync } from '../../../testing/async-helpers';
+import { richTextMentionUsersViewTag } from '../../../rich-text-mention/users/view';
+import {
+    RichTextMentionTest,
+    richTextMentionTestTag
+} from '../../../rich-text-mention/base/tests/rich-text-mention.fixtures';
+
+async function setup(): Promise<Fixture<RichTextEditor>> {
+    return fixture<RichTextEditor>(
+        html`<nimble-rich-text-editor></nimble-rich-text-editor>`
+    );
+}
+
+async function setupMentionConfig(): Promise<Fixture<RichTextEditor>> {
+    return fixture<RichTextEditor>(
+        // prettier-ignore
+        html`<${richTextEditorTag}>
+          <${richTextMentionUsersTag} pattern="^user:(.*)">
+              <${mappingUserTag} key="user:1" display-name="John Doe"></${mappingUserTag}>
+              <${mappingUserTag} key="user:2" display-name="Mary Wilson"></${mappingUserTag}>
+          </${richTextMentionUsersTag}>
+      </${richTextEditorTag}>`
+    );
+}
+
+async function appendUserMentionConfiguration(
+    element: RichTextEditor,
+    userKeys?: string[],
+    displayNames?: string[]
```

Comment:
It's confusing reading the tests that these are sperate properties. Instead could it be:

interface MappingConfiguration {
    key?: string
    displayname?: string
}

appendUserMentionConfiguration(
    element: RichTextEditor,
    mappings? MappingConfiguration[]
)

That way it's clearer what the mapping objects will look like as the related properties are grouped together instead of in separate arrays:
```
appendUserMentionConfiguration(
                element,
                [{key: 'user:1', displayName: 'username1'},
                {key: 'user:2', displayName: 'username2'}]
            );
```

Reply:
Updated